### PR TITLE
ft: #184361233 add payment using Stripe API

### PR DESCRIPTION
**Pull request by shyakadavis**
# Description

This P.R delivers the story #184361233, which is `Buyer should be able to make a payment using Stripe API` 💰

⚠️ 
- This is not test-able via the normal means, (ThunderClient, Swagger, or Postman), but you will need a working F.E to test the payment ft.)
- One thing to note too, is that we need to update the environment of the deployed API, and add the `STRIPE_SECRET`

Fixes # (issue)

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by SergeInGithub — COMMENTED** [review]
LGTM

**Review by SergeInGithub — DISMISSED** [review]
LGTM

File: src/controllers/Checkout.controller.ts [hunk 1]
```diff
@@ -6,20 +6,37 @@ const stripe = new Stripe(process.env.STRIPE_SECRET as string, {
...
-      // we still need ID coming from the front-end for this to work
-      const paymentIntent = await stripe.paymentIntents.create({
-        amount: 4000, // generic number
+      const customer = await stripe.customers.create({
+        email: req.body.email,
+        source: req.body.tokenId,
+        description: "Customer from the e-commerce",
+      });
+
+      const amount = req.body.amount;
```
Comment (mucyoalexaxel) [inline src/controllers/Checkout.controller.ts:17]: @shyakadavis great idea right here, . just have a few suggestions for better clean code check below: 
```suggestion
      const { email, tokenId, amount } = req.body;
      
      const customer = await stripe.customers.create({
        email,
        source: tokenId,
        description: "Customer from the e-commerce",
      });
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@shyakadavis Great work here. Check s few recommended changes.

File: src/controllers/Checkout.controller.ts (hunk 1 above)
Comment (shyakadavis) [reply src/controllers/Checkout.controller.ts:17]: Awesome. 🥏

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Review by SergeInGithub — DISMISSED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Pull request merged**